projects/flow.groups-2D-random-walk/src/project.py

```diff
@@ -0,0 +1,211 @@
+def all_simulated(*jobs):
+    """Return whether all jobs simulated."""
+    return all(simulated(job) for job in jobs)
+
+
+# Operations on individual jobs
+
+# Group for operations on single jobs that are reachable from simulate
+base = RandomWalkProject.make_group("base")
+
+
+@base
+@RandomWalkProject.post(simulated)
+@RandomWalkProject.operation
+def simulate(job):
+    """Simulate a 2D random walk."""
+    # use job id as random seed
+    n_steps = job.doc.run_steps
+    generator = np.random.default_rng(int(job.id, 16))
+    # Generate all moves at once since they are independent
+    moves = generator.normal(job.sp.mean, job.sp.standard_deviation, (n_steps, 2))
+    position = np.zeros((n_steps + 1, 2), dtype=float)
+    # Store the results: cumsum aggregates all previous moves from the origin
+    position[1:] = np.cumsum(moves, axis=0)
+    job.data.positions = position
+
+
+@base
+@RandomWalkProject.pre.after(simulate)
+@RandomWalkProject.post.true("radius_of_gyration")
+@RandomWalkProject.operation
+def compute_radius_of_gyration(job):
+    """Compute the radius of gyration for a random walk."""
+    with job.data:
+        positions = job.data["positions"][:]
+    avg_position = positions.mean(axis=0)
+    deviation = positions - avg_position
+    job.doc.radius_of_gyration = np.sqrt(
+        np.sum(deviation * deviation) / deviation.shape[0]
+    )
+    job.doc.average_position = avg_position
+
+
+@base
+@RandomWalkProject.pre.after(simulate)
+@RandomWalkProject.post.true("end_to_end")
+@RandomWalkProject.operation
+def compute_end_to_end_distance(job):
+    """Compute the radius of gyration for a random walk."""
+    with job.data:
+        positions = job.data["positions"]
+        job.doc["end_to_end"] = np.linalg.norm(positions[-1] - positions[0])
+
+
+@base
+@RandomWalkProject.pre.after(simulate)
+@RandomWalkProject.post(lambda job: "squared_distance" in job.data)
+@RandomWalkProject.operation
+def compute_squared_distance(job):
+    """Compute the squared distance for a random walk."""
+    with job.data:
+        positions = job.data.positions[:]
+    job.data.squared_distance = np.sum(positions * positions, axis=1)
+
+
+# This operation happens after computing the msd so it isn't in base.
+# Also we use pre as a filter for jobs ensuring this job only ever runs on the zeroth
+# replica for a given standard deviation.
+@RandomWalkProject.pre.true("msd_analyzed")
+@RandomWalkProject.pre(lambda job: job.sp.replica == 0)
+@RandomWalkProject.post.isfile("msd.png")
+@RandomWalkProject.operation
+def plot_msd(job):
+    """Plot the MSD for all std."""
+    with job.data:
+        msd = job.data.msd[:]
+    fig, ax = plt.subplots()
+    ax.plot(msd)
+    ax.set_title(f"MSD for {job.sp.standard_deviation} std")
+    ax.set_xlabel("x")
+    ax.set_ylabel("MSD")
+    # Only save figure to the first replica
+    fig.savefig(job.fn("msd.png"))
+    fig.close()
+
+
+# Aggregate operations on individual standard deviations across replicas
```

Comment:
I'm not sure that this comment adds anything to the content below, and I think the wording is actually more confusing.
```suggestion
```